presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexFactory.java

```diff
@@ -58,50 +87,64 @@ public PagesSpatialIndexFactory(List<Type> types, List<Type> outputTypes)
         return outputTypes;
     }
 
-    public synchronized ListenableFuture<PagesSpatialIndex> createPagesSpatialIndex()
+    /**
+     * Called by {@link SpatialJoinOperator} to get a Future that provides spatial
+     * index {@link Supplier} when done. {@link SpatialJoinOperator} must call
+     * {@link #probeOperatorFinished()} on completion to signal that spatial index
+     * is no longer needed.
+     */
+    public synchronized ListenableFuture<Supplier<PagesSpatialIndex>> createPagesSpatialIndex()
```

Comment:
as i noted in the operator, this could return `ListenableFuture<PagesSpatialIndex>`